crates/rpc/src/handlers/state.rs

```diff
@@ -1,40 +1,25 @@
-use std::{str::FromStr, sync::Arc};
-
-use alloy_primitives::B256;
 use ream_consensus::deneb::beacon_state::BeaconState;
-use ream_storage::{
-    db::ReamDB,
-    tables::{Table, beacon_state::BeaconStateTable, state_root_index::StateRootIndexTable},
-};
+use ream_storage::{db::ReamDB, tables::Table};
 
-use crate::types::errors::ApiError;
+use crate::types::{errors::ApiError, id::ID};
 
-pub async fn get_state_from_id(state_id: String, db: Arc<ReamDB>) -> Result<BeaconState, ApiError> {
-    let state_root = match B256::from_str(&state_id) {
-        Ok(value) => value,
-        Err(_) => {
-            return Err(ApiError::BadRequest(state_id));
-        }
+pub async fn get_state_from_id(state_id: ID, db: ReamDB) -> Result<BeaconState, ApiError> {
+    let state_root_result = match state_id {
+        ID::Slot(s) => db.slot_index_provider().get(s),
```

Comment:
```suggestion
        ID::Slot(slot) => db.slot_index_provider().get(slot),
```

please avoid 1 letter variables